src/wallet/spend.cpp

```diff
@@ -187,13 +187,15 @@ CoinsResult AvailableCoins(const CWallet& wallet,
 
             std::unique_ptr<SigningProvider> provider = wallet.GetSolvingProvider(output.scriptPubKey);
 
-            bool solvable = provider ? IsSolvable(*provider, output.scriptPubKey) : false;
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/24699/commits/45736de91e9350258994183c7032d51bf2e1142e: 

not sure if this is super important, but `IsSolvable` does a static_assert to check for `STANDARD_SCRIPT_VERIFY_FLAGS` and `SCRIPT_VERIFY_WITNESS_PUBKEYTYPE` and also runs `VerifyScript`, neither of which are happening in `CalculateMaximumSignedInputSize`

Reply:
maybe it's worth adding these checks to `DummySignInput` so there is no behavior change?

Reply:
`static_assert` is a compile time check so it doesn't matter where it is done.

Both `IsSolvable` and `CalculateMaximumSignedInputSize` call `ProduceSignature`, which does `VerifyScript` at the very end of it if the rest of it was successful. So `IsSolvable` doing `VerifyScript` is actually redundant, and both `IsSolvable` and `CalculatemaximumSignedInputSize` are doing the same checks.

Reply:
ah, i missed the `VerifyScript` call in `ProduceSignature`, thanks for the explanation!